libpod/healthcheck.go

```diff
@@ -397,15 +398,19 @@ func (c *Container) updateHealthCheckLog(hcl define.HealthCheckLog, inStartPerio
 			}
 		}
 	}
+	isHCStausChanged := oldHCStatus != healthCheck.Status
 	healthCheck.Log = append(healthCheck.Log, hcl)
 	if len(healthCheck.Log) > MaxHealthCheckNumberLogs {
 		healthCheck.Log = healthCheck.Log[1:]
 	}
 	newResults, err := json.Marshal(healthCheck)
 	if err != nil {
-		return "", fmt.Errorf("unable to marshall healthchecks for writing: %w", err)
+		return "", isHCStausChanged, fmt.Errorf("unable to marshall healthchecks for writing: %w", err)
+	}
+	if isHCStausChanged {
+		err = os.WriteFile(c.healthCheckLogPath(), newResults, 0700)
```

Comment:
That is not correct, the log contains timestamp and output of the command and that must be logged for each healthcheck as it is displayed in podman inspect.

Reply:
Got it. I thought  `healthCheckLogPath` is a persistent log of all `health_status` events. 
I will correct it once we finalize how we can move forward.

Reply:
done in https://github.com/containers/podman/pull/24005/commits/5b8d32d26bc1eebc45d82b40e9866a0459a6a500